sdk/communication/AzureCommunicationCommon/Tests/CommunicationIdentifierHelperTests.swift

```diff
@@ -0,0 +1,87 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCommunicationCommon
+import XCTest
+
+class CommunicationIdentifierHelperTests: XCTestCase {
+    func test_createUnknownIdentifier() {
+        var unknownIdentifier = CommunicationIdentifierHelper.createCommunicationIdentifier(from: "some id")
+        XCTAssertTrue(unknownIdentifier.isKind(of: UnknownIdentifier.self))
+
+        unknownIdentifier = CommunicationIdentifierHelper.createCommunicationIdentifier(from: "8:not_real_scope")
+        XCTAssertTrue(unknownIdentifier.isKind(of: UnknownIdentifier.self))
+
+        unknownIdentifier = CommunicationIdentifierHelper
+            .createCommunicationIdentifier(from: "8:not_real_scope:not_real_raw_id")
+        XCTAssertTrue(unknownIdentifier.isKind(of: UnknownIdentifier.self))
+    }
+
+    func test_createPhoneNumberIdentifier() {
+        var phoneNumberIdentifier = CommunicationIdentifierHelper.createCommunicationIdentifier(from: "4:11231234")
+        XCTAssertTrue(phoneNumberIdentifier.isKind(of: PhoneNumberIdentifier.self))
+        phoneNumberIdentifier = CommunicationIdentifierHelper.createCommunicationIdentifier(from: "4:11231234:asdf")
+
+        XCTAssertTrue(phoneNumberIdentifier.isKind(of: PhoneNumberIdentifier.self))
+    }
+
+    func test_createCommunicationUserIdentifier() {
+        let acsScope = "8:acs:some_raw_id"
+        var communciationUserIdentifier = CommunicationIdentifierHelper.createCommunicationIdentifier(from: acsScope)
+        XCTAssertTrue(communciationUserIdentifier.isKind(of: CommunicationUserIdentifier.self))
```

Comment:
Can we do a full object comparison?

Reply:
Are you thinking comparing every property inside `communciationUserIdentifier` and verifying that way